### PR TITLE
Fix/post pr 28707 comments

### PR DESCRIPTION
**Pull request by hicklin**
This PR addresses post merge comment for PR #28707. The comments addressed are

1. https://github.com/project-chip/connectedhomeip/pull/28707#discussion_r1309351951
2. https://github.com/project-chip/connectedhomeip/pull/28707#discussion_r1309352673

Apart from addressing these comments, this PR

1. Adds a check in the `UpdateAirQuality` method to ensure that the value given is not beyond the valid enum values.
2. Adds an out-of-band API to set the AirQuality attribute in the linux all-clusters-app.

## Testing

The out-of-band API was used to manually test that the `UpdateAirQuality` method works as expected. Different applications with one of the features switched off were built and it was confirmed that setting an enum corresponding to the unsupported feature fails and doesn't fail for the rest.

### CODE REVIEW

File: examples/all-clusters-app/linux/AllClustersCommandDelegate.cpp [hunk 1]
```diff
@@ -164,6 +165,19 @@ void AllClustersAppCommandHandler::HandleCommand(intptr_t context)
         }
         self->OnModeChangeHandler(device, type, mode);
     }
+    else if (name == "SetAirQuality")
+    {
+        Json::Value jsonAirQualityEnum = self->mJsonValue["Enum"];
```
Comment (bzbarsky-apple) [inline examples/all-clusters-app/linux/AllClustersCommandDelegate.cpp:170]:
```suggestion
        Json::Value jsonAirQualityEnum = self->mJsonValue["NewValue"];
```

File: examples/all-clusters-app/linux/AllClustersCommandDelegate.cpp [hunk 2]
```diff
@@ -164,6 +165,19 @@ void AllClustersAppCommandHandler::HandleCommand(intptr_t context)
...
+
+        if (jsonAirQualityEnum.isNull())
+        {
+            ChipLogError(NotSpecified, "The SetAirQuality command requires the Enum key.");
```
Comment (bzbarsky-apple) [inline examples/all-clusters-app/linux/AllClustersCommandDelegate.cpp:174]:
```suggestion
            ChipLogError(NotSpecified, "The SetAirQuality command requires the NewValue key.");
```

File: examples/all-clusters-app/linux/AllClustersCommandDelegate.cpp [hunk 3]
```diff
@@ -414,6 +428,18 @@ void AllClustersAppCommandHandler::OnModeChangeHandler(std::string device, std::
     }
 }
 
+void AllClustersAppCommandHandler::OnAirQualityChange(uint32_t aEnum)
```
Comment (bzbarsky-apple) [inline examples/all-clusters-app/linux/AllClustersCommandDelegate.cpp:431]:
```suggestion
void AllClustersAppCommandHandler::OnAirQualityChange(uint32_t aNewValue)
```
etc.

File: src/app/clusters/air-quality-server/air-quality-server.cpp [hunk 4]
```diff
@@ -62,28 +62,28 @@ Protocols::InteractionModel::Status Instance::UpdateAirQuality(AirQualityEnum aN
     switch (aNewAirQuality)
     {
     case AirQualityEnum::kFair: {
-        if (!HasFeature(AirQualityEnum::kFair))
+        if (!HasFeature(Feature::kFair))
```
Comment (bzbarsky-apple) [inline src/app/clusters/air-quality-server/air-quality-server.cpp:65]: Yikes, good thing you caught this!

File: src/app/clusters/air-quality-server/air-quality-server.cpp [hunk 5]
```diff
@@ -93,6 +93,11 @@ Protocols::InteractionModel::Status Instance::UpdateAirQuality(AirQualityEnum aN
         break;
     }
 
+    if (to_underlying(aNewAirQuality) > 6)
```
Comment (bzbarsky-apple) [inline src/app/clusters/air-quality-server/air-quality-server.cpp:96]: Is there a reason not to put this in the switch default case?

**Review by bzbarsky-apple — APPROVED** [review]

File: src/app/clusters/air-quality-server/air-quality-server.cpp (hunk 4 above)
Comment (hicklin) [reply src/app/clusters/air-quality-server/air-quality-server.cpp:65]: That was exactly my reaction!

File: src/app/clusters/air-quality-server/air-quality-server.cpp (hunk 5 above)
Comment (hicklin) [reply src/app/clusters/air-quality-server/air-quality-server.cpp]: Yeah. It's better in the switch. See [here](https://github.com/project-chip/connectedhomeip/pull/28993/commits/ca664fabb50221311c3930917c8ee3246fb88fb8).

**Review by tobiasgraf — APPROVED** [review]

**Review by p0fi — APPROVED** [review]

**Pull request merged**